pkg/machinery/extensions/validate.go

```diff
@@ -16,24 +16,83 @@ import (
 	"github.com/siderolabs/talos/pkg/machinery/version"
 )
 
-// Validate the extension: compatibility, contents, etc.
-func (ext *Extension) Validate() error {
-	if err := ext.validateConstraints(); err != nil {
-		return err
+// ValidationOptions are used to configure the validation process.
+type ValidationOptions struct {
+	// ValidateContstraints enables validation of the extension constraints.
+	ValidateContstraints bool
+	// ValidateContents enables validation of the extension contents.
+	ValidateContents bool
+
+	// TalosVersion is the version of Talos to validate against.
+	TalosVersion *semver.Version
+}
+
+// WithValidateConstraints enables validation of the extension constraints.
+func WithValidateConstraints() ValidationOption {
+	return func(o *ValidationOptions) error {
+		o.ValidateContstraints = true
+
+		return nil
+	}
+}
+
+// WithValidateContents enables validation of the extension contents.
+func WithValidateContents() ValidationOption {
+	return func(o *ValidationOptions) error {
+		o.ValidateContents = true
+
+		return nil
 	}
+}
+
+// WithTalosVersion sets the Talos version to validate against.
+func WithTalosVersion(version semver.Version) ValidationOption {
+	return func(o *ValidationOptions) error {
+		o.TalosVersion = &version
 
-	return ext.validateContents()
+		return nil
+	}
 }
 
-func (ext *Extension) validateConstraints() error {
-	constraint := ext.Manifest.Metadata.Compatibility.Talos.Version
+// ValidationOption is a function that configures the validation options.
+type ValidationOption func(*ValidationOptions) error
 
-	if constraint != "" {
-		talosVersion, err := semver.ParseTolerant(version.Tag)
+// Validate the extension: compatibility, contents, etc.
+func (ext *Extension) Validate(opts ...ValidationOption) error {
```

Comment:
did this, so `Validate` is backword compatible, I could also negate the options, so `Validate` works as is